### PR TITLE
Separate openapi check workflow from build one

### PR DESCRIPTION
**Pull request by mwithi**
This is an alternative solution to #421 in order to separate the `build` (may success) and `openapi` generation (may fail).

### CODE REVIEW

**Comment by mwithi** [comment]
@npasquetto after failing several attemps to have two subsequente jobs in the same worflow, I propose this version in place of #421 in order to separate the build (may success) and openapi generation (may fail).

Wdyt?

File: .github/workflows/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# This workflow will build and run the application for oasdiff/oasdiff-action/breaking@main
+# For more information see: https://github.com/oasdiff/oasdiff-action
+
+name: OpenAPI breaking changes check 
+
+on: [push, pull_request]
+
```
Comment (npasquetto) [inline .github/workflows/openapi.yml:7]: add:

```yaml
env:
  MAVEN_ARGS: '-B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
```
to reduce verbosity, and remove -q arg from mvn commands

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -11,8 +11,6 @@ jobs:
```
Comment (npasquetto) [inline .github/workflows/maven.yml:11]: add:

```yaml
env:
  MAVEN_ARGS: '-B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
```
to reduce verbosity, and remove -q arg from mvn commands

File: .github/workflows/openapi.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        with:
+          persist-credentials: false
+
+      - name: Checkout core
+        run: git clone --depth=50 --branch=develop https://github.com/informatici/openhospital-core.git openhospital-core
+
+      - name: Install core
+        run: cd openhospital-core && mvn -q install -DskipTests=true && cd ..
+
+      - name: Set up JDK 11
```
Comment (npasquetto) [inline .github/workflows/openapi.yml:23]: move this step before "Install core"

File: .github/workflows/maven.yml [hunk 4]
```diff
@@ -25,47 +23,5 @@ jobs:
         with:
```
Comment (npasquetto) [inline .github/workflows/maven.yml:23]: move this step before "Install core"

**Review by npasquetto — COMMENTED** [review]
I left some comments, consider whether it is appropriate to apply the proposed changes.

Unfortunately I was not able to successfully simulate the execution via the 'act' command but from the pipeline logs the workflow seems to have been successful.

**Comment by mwithi** [comment]
@npasquetto can we merge this PR?

**Pull request merged**